core/src/mender-client.c

```diff
@@ -655,65 +519,51 @@ mender_client_exit(void) {
         mender_client_artifact_types_list = NULL;
     }
     mender_client_artifact_types_count = 0;
-    mender_scheduler_mutex_give(mender_client_artifact_types_mutex);
-    mender_scheduler_mutex_delete(mender_client_artifact_types_mutex);
     mender_client_artifact_types_mutex = NULL;
     if (NULL != mender_client_addons_list) {
         free(mender_client_addons_list);
         mender_client_addons_list = NULL;
     }
     mender_client_addons_count = 0;
-    mender_scheduler_mutex_give(mender_client_addons_mutex);
-    mender_scheduler_mutex_delete(mender_client_addons_mutex);
     mender_client_addons_mutex = NULL;
 
-    return ret;
+    return MENDER_OK;
 }
 
 static mender_err_t
 mender_client_work_function(void) {
 
     mender_err_t ret = MENDER_OK;
 
-    /* Work depending of the client state */
-    if (MENDER_CLIENT_STATE_INITIALIZATION == mender_client_state) {
-        /* Perform initialization of the client */
-        if (MENDER_DONE != (ret = mender_client_initialization_work_function())) {
-            goto END;
-        }
-        /* Update client state */
-        mender_client_state = MENDER_CLIENT_STATE_AUTHENTICATION;
-    }
-    /* Request access to the network */
-    if (MENDER_OK != (ret = mender_client_network_connect())) {
-        goto END;
-    }
-    /* Intentional pass-through */
-    if (MENDER_CLIENT_STATE_AUTHENTICATION == mender_client_state) {
-        /* Perform authentication with the server */
-        if (MENDER_DONE != (ret = mender_client_authentication_work_function())) {
-            goto RELEASE;
-        }
-        /* Update work period */
-        if (MENDER_OK != (ret = mender_scheduler_work_set_period(mender_client_work_handle, mender_client_config.update_poll_interval))) {
-            mender_log_error("Unable to set work period");
-            goto RELEASE;
-        }
-        /* Update client state */
-        mender_client_state = MENDER_CLIENT_STATE_AUTHENTICATED;
-    }
-    /* Intentional pass-through */
-    if (MENDER_CLIENT_STATE_AUTHENTICATED == mender_client_state) {
-        /* Perform updates */
-        ret = mender_client_update_work_function();
-    }
-
-RELEASE:
+    mender_log_info("work function: %d", mender_client_state);
 
-    /* Release access to the network */
-    mender_client_network_release();
+    switch (mender_client_state) {
+        case MENDER_CLIENT_STATE_INITIALIZATION:
+            mender_log_info("work function; initialization");
+            if (MENDER_DONE != (ret = mender_client_initialization_work_function())) {
+                return ret;
+            }
+            /* Update client state */
+            mender_client_state = MENDER_CLIENT_STATE_AUTHENTICATION;
+            break;
```

Comment:
why `break` here? Can't we just continue without waiting for the next execution of this function hitting this `switch()` again?

Reply:
Yes, I'll do that.